aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py

```diff
@@ -49,168 +49,58 @@ def __init__(
 
-        :type subprocess_nodejs: aws_lambda_builders.workflows.nodejs_npm_esbuild.node.SubprocessNodejs
-        :param subprocess_nodejs: An instance of the nodejs process wrapper
-
         :type skip_deps: bool
         :param skip_deps: if dependencies should be omitted from bundling
 
         :type bundler_config: Dict[str,Any]
         :param bundler_config: the bundler configuration
+
+        :type manifest: str
+        :param manifest: path to package.json file contents to read
         """
         super(EsbuildBundleAction, self).__init__()
-        self.scratch_dir = scratch_dir
-        self.artifacts_dir = artifacts_dir
-        self.bundler_config = bundler_config
-        self.osutils = osutils
-        self.subprocess_esbuild = subprocess_esbuild
-        self.skip_deps = skip_deps
-        self.subprocess_nodejs = subprocess_nodejs
+        self._scratch_dir = scratch_dir
+        self._artifacts_dir = artifacts_dir
+        self._bundler_config = bundler_config
+        self._osutils = osutils
+        self._subprocess_esbuild = subprocess_esbuild
+        self._skip_deps = skip_deps
+        self._manifest = manifest
 
-    def execute(self):
+    def execute(self) -> None:
         """
         Runs the action.
 
         :raises lambda_builders.actions.ActionFailedError: when esbuild packaging fails
         """
+        esbuild_command = EsbuildCommandBuilder(
+            self._scratch_dir, self._artifacts_dir, self._bundler_config, self._osutils, self._manifest
+        )
 
-        explicit_entry_points = self._construct_esbuild_entry_points()
-
-        args = explicit_entry_points + ["--bundle", "--platform=node", "--format=cjs"]
-        minify = self.bundler_config.get("minify", True)
-        sourcemap = self.bundler_config.get("sourcemap", True)
-        target = self.bundler_config.get("target", "es2020")
-        external = self.bundler_config.get("external", [])
-        loader = self.bundler_config.get("loader", [])
-        if minify:
-            args.append("--minify")
-        if sourcemap:
-            args.append("--sourcemap")
-        if external:
-            args.extend(map(lambda x: f"--external:{x}", external))
-        if loader:
-            args.extend(map(lambda x: f"--loader:{x}", loader))
-
-        args.append("--target={}".format(target))
-        args.append("--outdir={}".format(self.artifacts_dir))
-
-        if self.skip_deps:
-            LOG.info("Running custom esbuild using Node.js")
-            # Don't pass externals because the esbuild.js template makes everything external
-            script = EsbuildBundleAction._get_node_esbuild_template(
-                explicit_entry_points, target, self.artifacts_dir, minify, sourcemap
-            )
-            self._run_external_esbuild_in_nodejs(script)
-            return
+        if self._should_bundle_deps_externally():
+            esbuild_command.build_with_no_dependencies()
```

Comment:
Instead of calling the external Node.js plugin to handle this, which would require us to update the Node.js template each time, we just make a list of all dependencies in the `package.json` file and set those to `external`.